### ISSUE TITLE
spacemacs used as GIT_EDITOR doesn't syntax highlight COMMIT_EDITMSG when magit isn't previously loaded

### ISSUE THREAD

**Issue by mkomitee** [syl20bnr/spacemacs#2522]
If you use emacs w/ spacemacs & magit via the git layer and emacsclient as GIT_EDITOR, and commit from a shell without first loading magit (by starting magit-status or something similar), then it won't syntax-highlight the COMMIT_EDITMSG buffer (it stays in FUNDAMENTAL mode) or enable the appropriate maps.

Adding (require 'magit) to dotspacemacs/config fixes the issue, so we probably need to add something else to the use-package stanza for magit in the git layer. I'd submit this as a pull request, but I'm not sure what would need to be added.

**Comment by mkomitee** [comment]
While most likely wrong (as per the magit docs), 

```
:mode ("COMMIT_EDITMSG" . git-commit-mode))
```

In the use-package directive for git-commit almost works, it just leaves evil in emacs state.

**Comment by syl20bnr** [comment]
I think this is fixed in develop thanks to https://github.com/syl20bnr/spacemacs/pull/2293

**Comment by mkomitee** [comment]
I'm running on `develop` already. To be clear, it works as GIT_EDITOR if I previously loaded magit by executing magit-status or magit-log (or any other command which causes use-package to require magit), but not if I haven't.

**Comment by StreakyCobra** [comment]
@mkomitee I'm using it in this way by having `(global-git-commit-mode t)` in the `user-config` of my `.spacemacs`. Does it work for you?

**Comment by synic** [comment]
@StreakyCobra It doesn't work for me.  In my `user-config`, I have this:

```elisp
  ;; Enable highlighting of git commit messages when emacs is $EDITOR
  (global-git-commit-mode t)
```

However, when running `git commit -a`, the commit message is in `Fundamental` mode.  

#### System Info
- OS: darwin
- Emacs: 24.5.7
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 7c160e8)
- Distribution: spacemacs
- Layers:
```elisp
(unscroll auto-completion xkcd emacs-lisp git gtags dash yaml github javascript html markdown extra-langs python ruby org osx python django syntax-checking spell-checking version-control lua colors spacemacs-layouts vimscript
          (c-c++ :variables c-c++-enable-clang-support t)
          (shell :variables shell-default-height 30 shell-default-position 'bottom))
```

**Comment by StreakyCobra** [comment]
Weird...

**Comment by synic** [comment]
Oh crap, nevermind.  I forgot I was running a daemon that needed reloading.  It does indeed work.

**Comment by mkomitee** [comment]
This appears to work for me. Should that be configured by the layer, or should everyone who uses spacemacs/emacs as `$EDITOR` have to set it themselves?

**Comment by StreakyCobra** [comment]
It should at least be documented :-) An option for this in the `git` layer would probably be better.

**Comment by StreakyCobra** [comment]
Fixed in 374e37e32c71295aec87d8207380e9ab019b5ba3

**Comment by mkomitee** [comment]
Thanks!
